kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/config/tls/NettyTrustProviderTest.java

```diff
@@ -47,15 +49,45 @@ void trustStoreTypes(String name, String storeType, String storeFile, PasswordPr
 
     @Test
     void trustStoreIncorrectPassword() {
-        var trustStore = new NettyTrustProvider(new TrustStore(TlsTestConstants.getResourceLocationOnFilesystem("client.jks"), TlsTestConstants.BADPASS, null));
+        var trustStore = new NettyTrustProvider(new TrustStore(TlsTestConstants.getResourceLocationOnFilesystem("client.jks"), TlsTestConstants.BADPASS, null, null));
         assertThatCode(() -> trustStore.apply(sslContextBuilder))
                 .hasMessageContaining("Error building SSLContext")
                 .hasRootCauseInstanceOf(UnrecoverableKeyException.class);
     }
 
+    @ParameterizedTest
+    @EnumSource(TlsClientAuth.class)
+    void clientAuthentication(TlsClientAuth clientAuth) {
+        var trustStore = new NettyTrustProvider(
+                new TrustStore(TlsTestConstants.getResourceLocationOnFilesystem("client.jks"), TlsTestConstants.STOREPASS, TlsTestConstants.JKS, clientAuth));
+        trustStore.apply(sslContextBuilder);
+        assertThat(sslContextBuilder)
+                .extracting("clientAuth")
+                .satisfies(nettyClientAuth -> {
+                    switch (clientAuth) {
+                        case REQUIRED -> assertThat(nettyClientAuth).isEqualTo(ClientAuth.REQUIRE);
+                        case REQUESTED -> assertThat(nettyClientAuth).isEqualTo(ClientAuth.OPTIONAL);
+                        case NONE -> assertThat(nettyClientAuth).isEqualTo(ClientAuth.NONE);
+                    }
```

Comment:
I understand why this looks like this, but conditionals in tests always make we worry about implementing the logic twice and thus the test is vulnerable to exactly the same issues as the production code. 

I know junit makes this annoying to implement but I'd either have three test cases where we can hard code the mappings or an argument source that pairs the `TlsClientAuth` to its `ClientAuth` counterpart and accept both as parameters. So that this simplifies to
```suggestion
                .isEqualTo(expectedClientAuth)
```

Reply:
Done.